Cleaned up dcarley's EBS work, adding documentation

blalor opened the pull request:
Still rough, but I really think the EBS stuff should be part of the main package. :-)

dcarley committed:
Remove ephemeral mounts

It's no longer as simple as determining the instance architecture.
Especially as this is the arch of the host instance rather than the guest.

Most AMIs handle this with an init process on boot that checks the meta-data
service for the instance's block mappings.

dcarley committed:
Add dual Grub menu entries for EBS and S3

Defaults to EBS. Falls back to S3. The only difference is the root
device - (hd0,0) for EBS and (hd0) for S3.

Reversed order of cfg and imgtemplate vars to match the real file.

dcarley committed:
Merge branch 'label_rootfs' into ebs

blalor committed:
Merge remote-tracking branch 'dcarley/ebs'

Conflicts:
	ami_creator/ami_creator.py

blalor committed:
Use Grub's hiddenmenu option

No need/ability to see the "graphical" menu anywhere.

blalor committed:
Attempt consistency in root filesystem label

There's something weird going on with the root filesystem label.
ExtDiskMount prefixes fslabel with "_".  It's not clear why, but in
trying to solve some problems booting the AMI built with this image I
was frustrated by the inconsistency and wanted to make it a little
more transparent (although I didn't make it any less dumb).

blalor committed:
Updated CentOS 6 KS config to my liking

Took some queues from Bashton's AMIs.

blalor committed:
capturing some docs and remaining steps

blalor committed:
Adding resource links

katzj commented:
I am game to do something along these lines and can take a closer look in the next couple of days.  I've got some related stuff that I use @ stackdriver to do similar so likely worth seeing where the overlaps are and aren't.

(PS glad to see that more people find this useful!)

blalor commented:
Ah, I thought your name sounded familiar.  We met at the BosOps meetup in your office a few weeks ago. :-)

Really, there's very little here except for the changes to the grub config.  I finished up a working script that will take a KS config, volume ID and block device path and registers an AMI.  I still find kickstart configs to be something of a black art with a disappointingly long cycle time for iterating on changes.  But this, coupled with Packer, should let me build AMIs and Vagrant boxes that are identical.

I'll add the script today.  You probably don't want to merge this PR directly.

blalor committed:
Adding helper scripts

create-ami.sh is the beaut.

blalor committed:
Allow for naming of AMI

blalor committed:
Hoping to make dd not exit zero without actually doing its job

blalor committed:
Work around some very wonky shit with dd

Commit:
added --virtualization-type optional parameter which feeds into register-image awscli call at the end of the script

Commit:
--kernel-id only set if virtualization_type is NOT hvm

Commit:
fixed device mapping issue for hvm instances

Commit:
adding grub patching and installation (needs to be done only if hvm is set, but for now lets just add it and get it working)

Commit:
if building an hvm ami: patch grub-install for xen device names, install grub, fix device.map, ..., profit

Commit:
create-ami.sh now prints out its args before executing, for debugging

blalor committed:
reformat register-image command

blalor committed:
Check ami_name for length

blalor committed:
patch is required (sadly)

blalor committed:
Don't rely on external source for diff

blalor committed:
Only patch grub-install once

Make this script more re-run-able

blalor committed:
Who knows what black magic has been performed on this volume previously!

blalor committed:
Normalize reference to ${block_dev}1

blalor committed:
fixup 676a4e8a4cbb01da10d18f7733cfe0a5b8f394cd

blalor committed:
fixup 64881ed83e5e767adfedc4802deecf3a4721c340

blalor committed:
Comments and hvm root device name

Per the block-device-mappings.html doc.

blalor committed:
Merge branch 'docs_n_tweaks' into bsdmaster

blalor committed:
Bail if AMI with same name already exists

blalor committed:
CentOS 7 prep

* Replace sfdisk with parted
* also generate some json

blalor committed:
Add utility scripts for building images in Docker

This allows for building images for distributions regardless of the
host you're building them on.

blalor committed:
Do not generate image, use pre-generated one

Plus other cleanup

blalor committed:
Add support for grub2

blalor committed:
Ensure image exists

blalor committed:
Ensure all required tools are installed

blalor committed:
Retry partition table re-reading

Saw a transient error…

blalor committed:
Include virtualization type in json

blalor committed:
Reduce verbosity

But only a little, really…

blalor committed:
Merge pull request #1 from bluestatedigital/master

Improved wrapper scripts
